packages/apollo-server-core/src/requestPipeline.ts

```diff
@@ -348,12 +348,18 @@ export async function processGraphQLRequest<TContext>(
     }
 
     if (config.formatResponse) {
-      response = config.formatResponse(response, {
-        context: requestContext.context,
-      });
+      const formattedResponse: GraphQLResponse | null = config.formatResponse(
+        response,
+        {
+          context: requestContext.context,
+        },
+      );
+      if (formattedResponse != null) {
```

Comment:
This isn't new in my PR, right?

Reply:
No, but it only surfaced because of the changes you made. It seems when a function (like `formatResponse`) returns `any`, the type gets widened to the assigned type (`GraphQLResponse | null`), losing the type narrowing that occurred before.